utility/nsfs_server_utils.py

```diff
@@ -224,10 +245,21 @@ def check_nsfs_tls_cert_setup(config_root):
         )
         return False
 
+    # Copy the remote TLS certificate to a temporary location where we can download it
+    tmp_tls_crt_file = "/tmp/tls.crt"
+    conn.exec_cmd(f"sudo cp {config_root}/certificates/tls.crt {tmp_tls_crt_file}")
+
+    # Check if the local and remote certificates match
+    comp_result = False
     with tempfile.TemporaryDirectory() as tmp_dir:
         local_tls_crt_file = f"{tmp_dir}/tls.crt"
         conn.download_file(
-            remotepath=f"{config_root}/certificates/tls.crt",
+            remotepath=tmp_tls_crt_file,
             localpath=f"{tmp_dir}/tls.crt",
         )
-        return compare_md5sums(local_tls_crt_file, S3Client.static_tls_crt_path)
+        comp_result = compare_md5sums(local_tls_crt_file, S3Client.static_tls_crt_path)
+
+    # Cleanup the temporary certificate file from the remote machine
+    conn.exec_cmd(f"sudo rm {tmp_tls_crt_file}")
+
+    return comp_result
```

Comment:
Just curious , Can we use reverse scp instead of all above operation

Reply:
I don't think so, because AFIK we can't elevate permissions via sudo in `scp`, in the same way we couldn't do it in `sftp` which is used here. 

However, while looking into your suggestion I got inspired with a more elegant workaround - just using `sudo ssh cat` and saving the output string to where we need to. Thanks to the way conn.exec_cmd is implemented this doesn't even spam the logs.

@udaysk23 please check the alternative solution that was added in https://github.com/red-hat-storage/noobaa-sa-ci/pull/29/commits/2297930c44b4ee7e298ddc23a85e31a677528df7 and let me know what you think.

Reply:
Verified [here](https://url.corp.redhat.com/516836e) after adding a small fix: https://github.com/red-hat-storage/noobaa-sa-ci/pull/29/commits/22e7fd3ebe0b60ced29dbd6c1e26fafa32c2714e